Principles/6 - Declarative over imperative.md

```diff
@@ -0,0 +1,27 @@
+# Declarative over imperative
+
+Imperative code is often a natural choice for developers who have spent a lot of time in C-style languages. The primitives of many popular programming languages are very easy to get started with writing highly imperative code. However, building imperative APIs can lead to much more error prone systems, more difficult to read code, and it being harder to discern the intention from the implementation for a given piece of logic.
```

Comment:
second sentence reads better as “the primitives of most programming languages make it easy to write imperative code”

Reply:
Switching to talking about imperative APIs feels weird, just stick to “code”

Reply:
Also avoid the “much”s and similar, the sentences are plenty strong without them

Reply:
I don’t feel we have convinced people why imperative code is more error prone/ harder to read

Reply:
Is it our job to convince in these? I don't know how we could do that without code examples.